plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -108,6 +110,17 @@ public synchronized TrinoCatalog create(ConnectorIdentity identity)
             icebergCatalog = icebergCatalogInstance;
         }
 
-        return new TrinoRestCatalog(icebergCatalog, catalogName, sessionType, trinoVersion, typeManager, uniqueTableLocation);
+        ImmutableMap.Builder<String, String> credentials = ImmutableMap.builder();
+        Optional.ofNullable(securityProperties.get().get(TOKEN)).ifPresent(token -> credentials.put(TOKEN, token));
+        Optional.ofNullable(securityProperties.get().get(CREDENTIAL)).ifPresent(credential -> credentials.put(CREDENTIAL, credential));
```

Comment:
Do you need this filtering or can you skip it and just pass `securityProperties.get()` through to  TrinoRestCatalog? This kinda breaks the encapsulation that the Factory here doesn't need to know what the contents of the map are.

Reply:
Good idea but I just filtered as securityProperties.get() may contain `OAuth2Properties.SCOPE` which is not used in 
https://github.com/apache/iceberg/blob/229d8f6fcd109e6c8943ea7cbb41dab746c6d0ed/core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java#L714-L721 rather it uses initially provided value of `OAuth2Properties.SCOPE`.